crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,927 @@
+    member: &JsStaticMemberExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    let args: Vec<_> = call_exp
+        .arguments()
+        .ok()?
+        .args()
+        .into_iter()
+        .flatten()
+        .collect();
+    if args.len() != 1 {
+        return None;
+    }
+    let AnyJsCallArgument::AnyJsExpression(arg0) = args[0].clone() else {
+        return None;
+    };
+    let core_arg0 = solve_parenthesized_expression(arg0)?;
+    let char_at_parent = solve_parenthesized_expression(member.object().ok()?)?;
+    match core_arg0.clone() {
+        // hoge.charAt(hoge.length - 1)
+        AnyJsExpression::JsBinaryExpression(_) => {
+            let at_number_exp =
+                extract_negative_index_expression(&core_arg0, char_at_parent.clone());
+            if let Some(at_number_exp) = at_number_exp {
+                Some(UseAtIndexState::new(
+                    at_number_exp,
+                    ErrorType::StringCharAt { is_negative: true },
+                    char_at_parent,
+                ))
+            } else {
+                option
+                    .check_all_index_access
+                    .then_some(UseAtIndexState::new(
+                        core_arg0,
+                        ErrorType::StringCharAt { is_negative: false },
+                        char_at_parent,
+                    ))
+            }
+        }
+        // hoge.charAt(1)
+        AnyJsExpression::AnyJsLiteralExpression(_member) => option
+            .check_all_index_access
+            .then_some(UseAtIndexState::new(
+                core_arg0,
+                ErrorType::StringCharAt { is_negative: false },
+                char_at_parent.clone(),
+            )),
+        _ => option
+            .check_all_index_access
+            .then_some(UseAtIndexState::new(
+                core_arg0,
+                ErrorType::StringCharAt { is_negative: false },
+                char_at_parent.clone(),
+            )),
+    }
+}
+
+/// check hoge.fuga()
+fn check_call_expression(
+    call_exp: &JsCallExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    // check slice
+    if let Some(slice_err) =
+        analyze_slice_element_access(&AnyJsExpression::JsCallExpression(call_exp.clone()))
+    {
+        return Some(slice_err);
+    }
+
+    if call_exp.is_optional_chain() {
+        return None;
+    }
+
+    let member = solve_parenthesized_expression(call_exp.callee().ok()?)?;
+    match member {
+        AnyJsExpression::JsStaticMemberExpression(member) => {
+            if member.is_optional_chain() {
+                return None;
+            }
+            let member_name = member
+                .member()
+                .ok()?
+                .as_js_name()?
+                .value_token()
+                .ok()?
+                .token_text_trimmed();
+            match member_name.text() {
+                "last" => check_call_expression_last(call_exp, &member),
```

Comment:
I think we should remove the lodash case for now. This looks too specific to me and could lead to many false positives.

```suggestion
```